### DISCUSSION TITLE
Manipulating Rasters: filters, get pixel value and customize colors,

### DISCUSSION THREAD

**Discussion by navilaufm** [opengeos/leafmap#288] [Q&A] [answered]
Hello, I am starting exploring Leafmap and has been able to add a geotiff to my map and set colors, but I am a bit confused how to add and set the legend.  and set the colors.

<img width="1296" alt="image" src="https://user-images.githubusercontent.com/61478833/191854427-771977d5-2b44-4d5a-bd1e-e8c1d85c875c.png">
See the following notebook:
https://colab.research.google.com/drive/1QdRPlZw_YTvXA6ScQZM4_jUcDCTRH85Z?usp=sharing

Its a geotiff with rainfall data  (24hr rainfall) 

1. I will like to not show the green for values less than 1 for example so the map can only show where it rained, like a 'filter' to just show values >1

2. Also I don't see an obvious way to set the color scale range so I guess I will plot max and min values for the pallete, I will like to manage as the colormap the max and min values so for example if I set 250 as the max, all above that will be RED even there could be values of 250 or 500 or 400. will show the same as are off the scale.

3. I will like to get the pixel value with a click or certain value.

Any ideas how to make these or any links to check on it ? 

thanks

**Comment by giswqs** [comment, marked as answer]
You should use the `add_colormap` function rather than the `add_colorbar` function

```python
m.add_colormap(cmap='RdYlGn_r',vmin=1,vmax=500,label='mm/24h')
```
![image](https://user-images.githubusercontent.com/5016453/192344305-c601f035-3d61-48c4-b545-4331331f80d7.png)
